### ISSUE TITLE
Using TypeVar of Generic type Can Not Pass Strict Check

### ISSUE THREAD

**Issue by gao462** [python/mypy#13820]
**Bug Report**

TypeVar of a Generic type is treated as Generic of Any, and this can not pass the check under `strict` mode (`disallow_any_generics = true`).

I think this usage should have a way to pass even under `strict` mode, e.g., keep typevar of generic still to be generic.

From general perspective, it seems to be related to this open thread: https://github.com/python/typing/issues/548, but I think this case is more commonly used, e.g., cloning.

**To Reproduce**

```python
from typing import Generic, TypeVar, List


T = TypeVar("T")
SelfBase = TypeVar("SelfBase", bound="Base")


class Base(Generic[T]):
    def __init__(self, mem: List[bytes]) -> None:
        self.mem = mem

    def __getitem__(self, i: int) -> T:
        pass

    def clone(self: SelfBase) -> SelfBase:
        return self.__class__(self.mem.copy())


class Int(Base[int]):
    def __getitem__(self, i: int) -> int:
        return int.from_bytes(self.mem[i], "big")


class Bits(Base[str]):
    def __getitem__(self, i: int) -> str:
        return bin(int.from_bytes(self.mem[i], "big"))[2:]


mem = [(i).to_bytes(1, "big") for i in range(8)]
decoder0 = Int(mem)
decoder1 = Bits(mem)
for i in range(8):
    print(decoder0[i], decoder1[i])
```

**Expected Behavior**

```bash
$ mypy test.py --strict
Success: no issues found in 1 source file
$ mypy test.py
Success: no issues found in 1 source file
```

**Actual Behavior**

```bash
$ mypy test.py --strict
test.py:5: error: Missing type parameters for generic type "Base"
Found 1 error in 1 file (checked 1 source file)
$ mypy test.py
Success: no issues found in 1 source file
```

**Your Environment**

- Mypy version used: 0.981
- Mypy command-line flags: (as in the behavior)
- Mypy configuration options from `mypy.ini` (and other config files): (no config files)
- Python version used: 3.10

**Comment by A5rocks** [comment]
Try `bound= "Base[Any]"`.
